boa/src/syntax/lexer/tests.rs

```diff
@@ -594,23 +595,32 @@ fn illegal_following_numeric_literal() {
 
     // Decimal Digit
     let mut lexer = Lexer::new(&b"11.6n3"[0..]);
-    assert!(
-        lexer.next().is_err(),
-        "DecimalDigit following NumericLiteral not rejected as expected"
-    );
+    match lexer.next() {
+        Err(Error::Syntax(_, pos)) => assert_eq!(pos, Position::new(1, 5)),
+        _ => assert!(
+            false,
+            "DecimalDigit following NumericLiteral not rejected as expected"
+        ),
+    }
```

Comment:
I think we should use `expect_err()` in these, makes things much cleaner:
```suggestion
    let err = lexer.next().expect_err("DecimalDigit following NumericLiteral not rejected as expected");
    if let Error::Syntax(_, pos) = err {
        assert_eq!(pos, Position::new(1, 5))
    } else {
        panic!("invalid error type");
    }
```